sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md

```diff
@@ -1,205 +1,170 @@
+    // Dispose of client.
+    client.close();
+  }
+```
 
-In v5, this has been consolidated into a more efficient `send(EventDataBatch)` method. Batching merges information from
-multiple events into a single sent message, reducing the amount of network communication needed vs sending events one at
-a time. Events are published to a specific partition [`CreateBatchOptions.setPartitionId()`][CreateBatchOptions] is set
-before calling `createBatch(CreateBatchOptions)`.
+### Send events to an Event Hub
 
-The code below assumes all events fit into a single batch. For a more complete example, see sample: [Publishing events
-to specific partition][PublishEventsToSpecificPartition].
+In v3, events could be published to a single partition using `PartitionSender` or automatically routed to an available partition via `EventHubClient`.
 
-So in v3:
-```java
-EventHubClient client = EventHubClient.createFromConnectionStringSync(
-    "connection-string-for-an-event-hub", Executors.newScheduledThreadPool(4));
-List<EventData> events = Arrays.asList(EventData.create("foo".getBytes()));
+In v5, the **same client** can create an [EventDataBatchs][EventDataBatch] that routes to a specific partition or use automatic routing. Events are published to a specific partition using [`CreateBatchOptions.setPartitionId()`][CreateBatchOptions]; otherwise, the batch is automatically routed.
 
-CompletableFuture<Void> sendFuture = client.createPartitionSender("my-partition-id")
-    .thenCompose(producer -> {
-        EventDataBatch batch = producer.createBatch();
-        // Assuming all events fit into a single batch. This returns false if it does not.
-        // If it returns false, we send the batch, create another, and continue to add events.
-        for (EventData event : events) {
-            try {
-                batch.tryAdd(event);
-            } catch (PayloadSizeExceededException e) {
-                System.err.println("Event is larger than maximum allowed size. Exception: " + e);
-            }
-        }
+Additionally, we consolidate send into an efficient `send(EventDataBatch)` method on the producer client. Batching merges information from multiple events into a single sent message, reducing the amount of network communication needed vs sending events one at a time.
 
-        return producer.send(batch);
-    });
+The code below assumes all events fit into a single batch. For a more complete example, see sample: [Publishing events to specific partition][PublishEventsToSpecificPartition].
 
-sendFuture.get();
-```
+So in v3:
 
-In v5:
 ```java
 List<EventData> events = Arrays.asList(EventData.create("foo".getBytes()));
 
-EventHubProducerAsyncClient producer = new EventHubClientBuilder()
-    .connectionString("connection-string-for-an-event-hub")
-    .buildAsyncProducerClient();
-
-CreateBatchOptions options = new CreateBatchOptions()
-    .setPartitionId("my-partition-id");
+ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(4);
```

Comment:
There is a mix of sync and async methods in these code snippets. Can we stick to just sync going by the assumption that most Java users would be using sync?

Reply:
I opted to use only async clients for my snippets because the migration guides are assuming these are in production. sync clients are what we suggest for getting started.